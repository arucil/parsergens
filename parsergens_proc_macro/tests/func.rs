
use insta::assert_debug_snapshot;

parsergens_proc_macro::parsergen!(mod clr_parser_with_func: LR("fixtures/expr_func.pg"));

#[test]
fn clr() {
  let input = r"factorial(7) * 1000 + gcd(12, 39)";
  let result = clr_parser_with_func::with_input(input)
    .parse_expr();
  assert_debug_snapshot!(result);
}

parsergens_proc_macro::parsergen!(mod lalr_parser_with_func: LALR("fixtures/expr_func.pg"));

#[test]
fn lalr() {
  let input = r"factorial(7) * 1000 + gcd(12, 39)";
  let result = lalr_parser_with_func::with_input(input)
    .parse_expr();
  assert_debug_snapshot!(result);
}