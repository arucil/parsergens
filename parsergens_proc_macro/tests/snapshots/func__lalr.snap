---
source: parsergens_proc_macro/tests/func.rs
expression: result
---
Ok(
    Ok(
        5040003.0,
    ),
)
